### PR TITLE
Add target platform version to "vs/solution/project/used"

### PR DESCRIPTION
**Pull request by ocallesp**
on the VS Win side, it needs to include the version for the target platform.

Before 
•	vs.solution.project.targetplatformmonikers="ios;android;windows"
After
•	vs.solution.project.targetplatformmonikers="ios;android;windows10.0.19041.0"

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/7394)

### CODE REVIEW

**Review by melytc — APPROVED** [review]

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/TargetPlatformMonikersValueProvider.cs [hunk 1]
```diff
@@ -34,8 +34,10 @@ public override async Task<string> OnGetEvaluatedPropertyValueAsync(string prope
                 ProjectProperties projectProperties = configuredProject.Services.ExportProvider.GetExportedValue<ProjectProperties>();
                 ConfigurationGeneral configuration = await projectProperties.GetConfigurationGeneralPropertiesAsync();
                 string? currentPlatformMoniker = (string?)await configuration.TargetPlatformIdentifier.GetValueAsync();
+                string? currentPlatformVersion = (string?)await configuration.TargetPlatformVersion.GetValueAsync();
+
                 Assumes.NotNull(currentPlatformMoniker);
-                builder.Add(currentPlatformMoniker);
+                builder.Add(currentPlatformMoniker + currentPlatformVersion);
```
Comment (jjmew) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/TargetPlatformMonikersValueProvider.cs:40]: do you need the Assumes.NotNull() for currentPlatformVersion?

Comment (ocallesp) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/TargetPlatformMonikersValueProvider.cs:40]: Not needed. If there is no version that is fine. We are printing the platform moniker and the version as it is.

Comment (jjmew) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/TargetPlatformMonikersValueProvider.cs:40]: Got it

**Comment by mrward** [comment]
What is the TargetPlatformVersion here? Is this the msbuild resolved value or the original version used in the TargetFramework or TargetFrameworks property?

**Review by jjmew — APPROVED** [review]
:shipit:

**Comment by ocallesp** [comment]
> What is the TargetPlatformVersion here? Is this the msbuild resolved value or the original version used in the TargetFramework or TargetFrameworks property?

This is the evaluated (resolved) property.

**Comment by mrward** [comment]
Did you test what TargetPlatformVersion returns for iOS or Android? With MAUI, for example, the resolved TargetPlatformVersion value returns 0.0 which is not the original string used in the TargetFrameworks property. In the past I have also seen this resolve to some default version (e.g. 10.0 for Android).

**Comment by mrward** [comment]
Looking at the iOS workload sdk I have installed locally, I see:

```
pwd
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk

grep "<TargetPlatformVersion" . -r
./14.5.100-preview.5.894/targets/Xamarin.Shared.Sdk.TargetFrameworkInference.props:    <TargetPlatformVersion Condition="'$(TargetPlatformVersion)' == ''">$(_DefaultTargetPlatformVersion)</TargetPlatformVersion>

grep "_DefaultTargetPlatformVersion" . -r
./14.5.100-preview.5.894/targets/Microsoft.iOS.Sdk.Versions.props:		<_DefaultTargetPlatformVersion>13.6</_DefaultTargetPlatformVersion>
./14.5.100-preview.5.894/targets/Xamarin.Shared.Sdk.TargetFrameworkInference.props:    <TargetPlatformVersion Condition="'$(TargetPlatformVersion)' == ''">$(_DefaultTargetPlatformVersion)</TargetPlatformVersion>
```

So it seems that it would resolve to some default value not the original string used.

I was planning to use the original string in VS Mac because of this problem.

**Review by mrward — APPROVED** [review]

**Pull request merged**